Example/Auth/Sample/MainViewController+Apple.m

```diff
@@ -0,0 +1,129 @@
+/*
+* Copyright 2019 Google
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#import "MainViewController+Apple.h"
+
+#import <AuthenticationServices/AuthenticationServices.h>
+
+#import "AppManager.h"
+#import "FirebaseAuth.h"
+#import "MainViewController+Internal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MainViewController () <ASAuthorizationControllerDelegate, ASAuthorizationControllerPresentationContextProviding> {
+
+}
+
+@end
+
+@implementation MainViewController (Apple)
+
+- (StaticContentTableViewSection *)appleAuthSection API_AVAILABLE(ios(13.0)) {
+  __weak typeof(self) weakSelf = self;
+  return [StaticContentTableViewSection sectionWithTitle:@"Apple Auth" cells:@[
+    [StaticContentTableViewCell cellWithTitle:@"Sign in with Apple"
+                                       action:^{ [weakSelf signInWithApple]; }],
+    [StaticContentTableViewCell cellWithTitle:@"Link with Apple"
+                                       action:^{ [weakSelf linkWithApple]; }],
+    [StaticContentTableViewCell cellWithTitle:@"Unlink with Apple"
+                                       action:^{ [weakSelf unlinkFromProvider:@"apple.com" completion:nil]; }],
+    [StaticContentTableViewCell cellWithTitle:@"Reauthenticate with Apple"
+                                       action:^{ [weakSelf reauthenticateWithApple]; }],
+  ]];
+}
+
+- (void)signInWithApple API_AVAILABLE(ios(13.0)) {
+  ASAuthorizationAppleIDProvider* provider = [[ASAuthorizationAppleIDProvider alloc] init];
+  ASAuthorizationAppleIDRequest* request = [provider createRequest];
+  request.requestedScopes = @[ASAuthorizationScopeEmail, ASAuthorizationScopeFullName];
+  request.nonce = @"c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2";
```

Comment:
This is not great nonce practice, maybe add a comment telling users to not do this in production.

Also, consider deduplicating the request config boilerplate into a separate method. Something like
```objc
- (ASAuthorizationAppleIDRequest *)appleIDRequestWithState:(NSString *)state {
  ASAuthorizationAppleIDRequest *request = [[[ASAuthorizationAppleIDProvider alloc] init] createRequest];
  request.requestedScopes = @[ ... ];
  request.nonce = @"";
  request.state = state;
  return request;
}
```

Reply:
Yeah, good idea.